R/bookmark-state.R

```diff
@@ -346,16 +346,22 @@ RestoreContext <- R6Class("RestoreContext",
       values <- parseQueryString(valueStr, nested = TRUE)
 
       valuesFromJSON <- function(vals) {
-        mapply(names(vals), vals, SIMPLIFY = FALSE,
+        valsParsed <- mapply(names(vals), vals, SIMPLIFY = FALSE,
           FUN = function(name, value) {
             tryCatch(
               safeFromJSON(value),
               error = function(e) {
-                stop("Failed to parse URL parameter \"", name, "\"")
+                # If a value did not parse from JSON successfully, return NA
+                # because that's a unique value that can't be obtained from JSON
+                NA
```

Comment:
I think you'll need a different sentinel value, since it is possible to get `NA`:
 
```R
safeFromJSON('[null]')
#> [1] NA
```

Reply:
@wch I suppose NA_character_ would be unachievable with jsonlite::fromJSON. But perhaps this is just too hacky and not the correct approach.